front/extension/app/src/pages/ConversationPage.tsx

```diff
@@ -1,16 +1,51 @@
-import { BarHeader, ChevronLeftIcon } from "@dust-tt/sparkle";
-import { Link } from "react-router-dom";
+import { useAuth } from "@app/extension/app/src/components/auth/AuthProvider";
+import { ConversationContainer } from "@app/extension/app/src/components/conversation/ConversationContainer";
+import { Page, Spinner } from "@dust-tt/sparkle";
+import { useNavigate, useParams } from "react-router-dom";
 
 export const ConversationPage = () => {
+  const navigate = useNavigate();
+  const { isLoading, isAuthenticated, user } = useAuth();
+  const { conversationId } = useParams();
+
+  console.log(conversationId);
+
+  if (isLoading) {
+    return (
+      <div className="h-full w-full">
+        <div className="flex h-full w-full items-center justify-center">
+          <Spinner />
+        </div>
+      </div>
+    );
+  }
+
+  if (!isAuthenticated || !user) {
+    navigate("/login");
+    return;
+  }
+
+  const workspace = user.workspaces.find(
+    (w) => w.sId === user.selectedWorkspace
+  );
+
+  if (!workspace) {
+    navigate("/login");
+    return;
+  }
+
+  if (!conversationId) {
+    navigate("/");
+    return;
+  }
+
   return (
-    <BarHeader
-      title="Conversations"
-      leftActions={
-        <Link to="/main.html">
-          <ChevronLeftIcon />
-          You're in the conversation page!
-        </Link>
-      }
-    />
```

Comment:
It's not pretty it goes on top of our header. I added it back anyway since that's an easy way for us to go back but we will need to improve the design!